conv_rnn/train.py

```diff
@@ -0,0 +1,192 @@
+            i += mbatch_size
+            if i % (mbatch_size * 256) == 0:
+                conv_rnn.eval()
+                dev_in, dev_out = conv_rnn.convert_dataset(dev_set)
+                scores = conv_rnn(dev_in)
+                n_correct = (torch.max(scores, 1)[1].view(len(dev_set)).data == dev_out.data).sum()
+                accuracy = n_correct / len(dev_set)
+                #scheduler.step(accuracy)
+                if accuracy > conv_rnn.best_dev:
+                    conv_rnn.best_dev = accuracy
+                    torch.save(conv_rnn, kwargs["output_file"])
+                if verbose:
+                    print("Dev set accuracy: {}".format(accuracy))
+                conv_rnn.train()
+    return conv_rnn.best_dev
+
+def do_random_search():
+    test_grid = [[0.15, 0.2], [4, 5, 6], [10, 20], [150, 200], [3, 4, 5], [200, 300], [200, 250]]
+    max_params = None
+    max_acc = 0.
+    for sf, gc, ptce, hid, seed, fc_size, fmaps in RandomSearch(test_grid):
+        print("Testing {}".format([sf, gc, ptce, hid, seed, fc_size, fmaps]))
+        dev_acc = train(mbatch_size=64, n_epochs=7, verbose=False, restore=False, gradient_clip=gc,
+                schedule_factor=sf, patience=ptce, hidden_size=hid, seed=seed, n_feature_maps=fmaps, 
+                fc_size=fc_size)
+        print("Dev accuracy: {}".format(dev_acc))
+        if dev_acc > max_acc:
+            print("Found current max")
+            max_acc = dev_acc
+            max_params = [sf, gc, ptce, hid, seed, fc_size, fmaps]
+    print("Best params: {}".format(max_params))
+
+def do_grid_search():
+    test_grid = [[0.6, 0.4, 0.2], [6, 7, 8], [1E-2, 1E-3, 1E-4]]
+    max_params = None
+    max_acc = 0.
+    for schedule_factor, gradient_clip, weight_decay in GridSearch(test_grid):
+        print("Testing {}".format([schedule_factor, gradient_clip, weight_decay]))
+        dev_acc = train(mbatch_size=64, n_epochs=7, verbose=False, restore=False,
+                gradient_clip=gradient_clip, schedule_factor=schedule_factor, weight_decay=weight_decay)
+        print("Dev accuracy: {}".format(dev_acc))
+        if dev_acc > max_acc:
+            max_acc = dev_acc
+            max_params = [schedule_factor, gradient_clip, weight_decay]
+    print("Best params: {}".format(max_params))
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--mbatch_size", type=int)
+    parser.add_argument("--n_epochs", type=int)
+    parser.add_argument("--restore", default=False, action="store_true")
+    parser.add_argument("--lr", type=float)
+    parser.add_argument("--weight_decay", type=float)
+    parser.add_argument("--gradient_clip", type=float)
+    parser.add_argument("--seed", type=int)
+    parser.add_argument("--hidden_size", type=int)
+    parser.add_argument("--fc_size", type=int)
+    parser.add_argument("--dropout_prob", type=float)
+    parser.add_argument("--n_feature_maps", type=float)
+    parser.add_argument("--rnn_type", type=str)

```

Comment:
Use `choices` keyword argument to specify what are the possible choices - see argparse docs for details.